### PR TITLE
add faiss debug option

### PR DESCRIPTION
**Pull request by datelier**
### Description:
SSIA

### Versions:

- Go Version: 1.20.5
- Docker Version: 20.10.8
- Kubernetes Version: v1.27.3

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

### CODE REVIEW

File: internal/core/algorithm/faiss/Capi.cpp [hunk 1]
```diff
@@ -31,9 +31,11 @@ FaissStruct* faiss_create_index(
...
     const int nbits_per_idx,
     const int metric_type) {
-  //printf(__FUNCTION__);
-  //printf("\n");
-  //fflush(stdout);
+#if DEBUG
+  printf(__FUNCTION__);
+  printf("\n");
+  fflush(stdout);
+#endif
```
Comment (kmrmt) [inline internal/core/algorithm/faiss/Capi.cpp:38]:
```suggestion
#if DEBUG
  printf("%s\n", __FUNCTION__);
  fflush(stdout);
#endif
```
You can use some macros such as `__PRETTY_FUNCTION__` and `__LINE__` for detail debug message.

FYI, there is one of some debugging techniques like below.
```
#ifdef DEBUG
#define DEBUG_MESSAGE(x) do { std::cout << "[" << __FUNCTION__ << ":" << __LINE__ << "]" << x << std::endl; } while(0)
#else
#define DEBUG_MESSAGE(x)
#endif
```

**Review by kmrmt — APPROVED** [review]
LGTM and I have some C++ techniques.

File: Makefile.d/build.mk [hunk 2]
```diff
@@ -70,9 +70,9 @@ cmd/agent/core/faiss/faiss: \
 	CFLAGS="$(CFLAGS)" \
 	CXXFLAGS="$(CXXFLAGS)" \
 	CGO_ENABLED=1 \
-	CGO_CXXFLAGS="-g -Ofast -march=native" \
-	CGO_FFLAGS="-g -Ofast -march=native" \
-	CGO_LDFLAGS="-g -Ofast -march=native" \
+	CGO_CXXFLAGS="-g -Ofast -march=native -DDEBUG=0" \
+	CGO_FFLAGS="-g -Ofast -march=native -DDEBUG=0" \
+	CGO_LDFLAGS="-g -Ofast -march=native -DDEBUG=0" \
```
Comment (ykadowak) [inline Makefile.d/build.mk:75]:
```suggestion
	CGO_CXXFLAGS="-g -Ofast -march=native -DDEBUG" \
	CGO_FFLAGS="-g -Ofast -march=native -DDEBUG" \
	CGO_LDFLAGS="-g -Ofast -march=native -DDEBUG" \
```

**Review by ykadowak — COMMENTED** [review]
Also it's better to do something like this once and use the typed variable for the rest.
```c++
#if defined(DEBUG)
constexpr bool isDebug = true;
#else
constexpr bool isDebug = false;
#endif
```

**Review by ykadowak — APPROVED** [review]

**Review by kevindiu — APPROVED** [review]

**Pull request merged**